.github/actions/docker-publish/action.yml

```diff
@@ -0,0 +1,46 @@
+name: "Publish Image to Docker Hub"
+description: |
+  Build and push the latest docker image to docker hub.
+  Three images will be pushed, `latest` and two semver images.
+  One will be the full semver image (major.minor.patch).
+  The second will be just the major.minor version.
+inputs:
+  docker_username:
+    required: true
+    description: "secrets.DOCKER_USERNAME"
+  docker_password:
+    required: true
+    description: "secrets.DOCKER_PASSWORD"
+  docker_repository:
+    required: true
+    description: "name of the docker repository"
+  semver_version:
+    required: true
+    description: "The semver version to release (vX.X.X)"
```

Comment:
I think the version should come from tag that triggered the build and not manual input, otherwise docker/setup-buildx-action might check out a different version than you intend to build (read more https://github.com/docker/build-push-action#git-context)